memcache/memcache.go

```diff
@@ -144,9 +150,12 @@ type Client struct {
 	// be set to a number higher than your peak parallel requests.
 	MaxIdleConns int
 
+	// number of currently used connections
+	activeConns int32
+
 	selector ServerSelector
 
-	lk       sync.Mutex
+	lk       sync.RWMutex
```

Comment:
doesn't seem worth it.

Reply:
maybe more straightforward to just make idleConns an atomic int as well and inc/dec when `freeconn` changes?